transaction/_transaction.py

```diff
@@ -138,6 +138,8 @@ def user(self):
     def user(self, v):
         if v is not None:
             self._user = text_or_warn(v)
+        else:
```

Comment:
If it isn't an error, then it should have a predictable outcome:  leaving whatever was in place before seems like a bug magnet.